arduino-ide-extension/src/browser/boards/boards-toolbar-item.tsx

```diff
@@ -138,7 +138,7 @@ export class BoardsDropDown extends React.Component<BoardsDropDown.Props> {
             {boardLabel}
           </div>
           <div className="arduino-boards-dropdown-item--port-label noWrapInfo noselect">
-            {port.addressLabel}
+            {port.addressLabel || port.address}
```

Comment:
maybe this is not appropriate, I imagine it depends on whether or not `.addressLabel` was intended to be a "catch all" or not, for the sake of efficiency I've plugged this quick fix with a question: @kittaakos are you able to confirm if a fallback is appropriate here, or should `port.addressLabel` be our only reference as asserted in [this PR](https://github.com/arduino/arduino-ide/pull/1335) Thanks!

Reply:
Were you able to determine why `port.addressLabel` is empty after opening a new window @davegarthsimpson?

Reply:
> Were you able to determine why port.addressLabel is empty after opening a new window
Yes, thanks for the important question, it's due to a chunk of code in our `reconcileAvailableBoards` method of the `boards-service-provider` [here](https://github.com/arduino/arduino-ide/blob/a6a7af4d44f3b6ad847748c267869671cdc8e6d8/arduino-ide-extension/src/browser/boards/boards-service-provider.ts#L592) where we set the port copying the `boardsConfig.selectedPort` object which does not contain `.addressLabel` only `.address`, [this](https://github.com/arduino/arduino-ide/blob/a6a7af4d44f3b6ad847748c267869671cdc8e6d8/arduino-ide-extension/src/browser/boards/boards-service-provider.ts#L581) comment explains the reason why the author wrote this logic.

Reply:
The other alternative would be something like this in the "culprit code":

before:
```
581     ...
        const found = availableBoards.findIndex(
           (board) => board.port?.address === boardsConfig.selectedPort?.address
         );
         if (found >= 0) {
           availableBoards.splice(found, 1);
         }
         availableBoards.push({
           ...boardsConfig.selectedBoard,
           port: boardsConfig.selectedPort,
           selected: true,
           state: AvailableBoard.State.incomplete,
         });
```

after:
```
581     let port = boardsConfig.selectedPort
         ...
         const found = availableBoards.findIndex(
           (board) => board.port?.address === boardsConfig.selectedPort?.address
         );
         if (found >= 0) {
           port = availableBoards[found].port // get the "Unknown board port" that we will substitute
           availableBoards.splice(found, 1);
         }
         availableBoards.push({
           ...boardsConfig.selectedBoard,
           port, // "copy" the port info here where we'll have an "addressLabel" too
           selected: true,
           state: AvailableBoard.State.incomplete,
         });
```

Reply:
Nice catch; the fix looks good to me. The breaking change was probably [here](https://github.com/arduino/arduino-ide/commit/af33dce0f63d219ee8c49ad37b0501afb1f78565#diff-dde17ee2d638b996314391e855eb70b94c0342f5e22dbff91c04d744d544a962R146-R153) when new props were added to the `Port` but on `board#port`.